user_guide_src/source/database/query_builder.rst

```diff
@@ -878,6 +880,12 @@ The first parameter is an associative array of values.
 
 .. warning:: When you use ``RawSql``, you MUST escape the data manually. Failure to do so could result in SQL injections.
 
+You can also insert from a query:
+
+.. literalinclude:: query_builder/117.php
+
+.. note:: It is required to alias the columns of the select query to match those of the target table. 
```

Comment:
Please add a note that `setQueryAsData()` can be used since v4.3.0.
```
.. note:: ``$setQueryAsData()`` can be used since v4.3.0.
```

This is necessary to ensure that users of any version can take advantage of the latest documentation.

Or if you add a section for a new item, you can use the tag:
```
.. versionadded:: 4.3.0
```